document/src/main/java/com/yahoo/document/json/JsonReader.java

```diff
@@ -97,28 +101,52 @@ public ParsedDocumentOperation readSingleDocumentStreaming(DocumentOperationType
             if (JsonToken.START_OBJECT != parser.nextValue())
                 throw new IllegalArgumentException("expected start of root object, got " + parser.currentToken());
 
-            parser.nextValue();
-            if ( ! FIELDS.equals(parser.getCurrentName()))
-                throw new IllegalArgumentException("expected field \"fields\", but got " + parser.getCurrentName());
+            Boolean create = null;
+            String condition = null;
+            ParsedDocumentOperation operation = null;
+            while (JsonToken.END_OBJECT != parser.nextValue()) {
+                switch (parser.getCurrentName()) {
+                    case FIELDS -> {
+                        documentParseInfo.fieldsBuffer = new LazyTokenBuffer(parser);
+                        VespaJsonDocumentReader vespaJsonDocumentReader = new VespaJsonDocumentReader(typeManager.getIgnoreUndefinedFields());
+                        operation = vespaJsonDocumentReader.createDocumentOperation(
+                                getDocumentTypeFromString(documentParseInfo.documentId.getDocType(), typeManager), documentParseInfo);
+
+                        if ( ! documentParseInfo.fieldsBuffer.isEmpty())
+                            throw new IllegalArgumentException("expected all content to be consumed by document parsing, but " +
+                                                               documentParseInfo.fieldsBuffer.nesting() + " levels remain");
+
+                    }
+                    case CONDITION -> {
+                        if ( ! JsonToken.VALUE_STRING.equals(parser.currentToken()) && ! JsonToken.VALUE_NULL.equals(parser.currentToken()))
+                            throw new IllegalArgumentException("expected string value for condition, got " + parser.currentToken());
+
+                        condition = parser.getValueAsString();
+                    }
+                    case CREATE_IF_NON_EXISTENT -> {
+                        create = parser.getBooleanValue(); // Throws if not boolean.
+                    }
+                    default -> {
+                        // We ignore stray fields, but need to ensure structural balance in doing do.
+                        if (parser.currentToken().isStructStart()) parser.skipChildren();
+                    }
```

Comment:
`"create"` and `"condition"` fields now required to be of the correct type, instead of mismatches being silently ignored, as before. 
Other fields are still silently ignored. 
Also, I don't like that create and condition can be supplied both here, and through HTTP headers, for `/document/v1/`.